piet-wgsl/src/shaders/preprocess.rs

```diff
@@ -71,29 +87,53 @@ pub fn preprocess(
                             item.active = !item.active;
                         }
                     }
+                    let remainder = directive_start[directive_len..].trim();
+                    if !remainder.is_empty() {
+                        eprintln!("#else directives don't take an argument. `{remainder}` will not be in output (line {line_number})");
+                    }
+                    continue 'outer;
                 }
                 "endif" => {
                     if let None = stack.pop() {
                         eprintln!("Mismatched endif (line {line_number})");
                     }
+                    continue 'outer;
                 }
                 "import" => {
-                    let import = imports.get(arg);
+                    output.push_str(&line[..hash_index]);
+                    let directive_end = &directive_start[directive_len..];
+                    let import_name_start = if let Some(import_name_start) =
+                        directive_end.find(|c: char| !c.is_whitespace())
+                    {
+                        import_name_start
+                    } else {
+                        eprintln!("#import needs a non_whitespace argument (line {line_number})");
+                        continue 'outer;
+                    };
+                    let import_name_start = &directive_end[import_name_start..];
+                    let import_name_end_index = import_name_start
```

Comment:
It seems to me this is doing the same logic as the calculation of directive_len but in a different way. I find this version slightly more intuitive and less fragile. Either way, unless there's a compelling reason it seems like the two should be the same.

Reply:
Well spotted, I've used the same technique for `directive_len ` now